packages/opentelemetry-basic-tracer/src/BasicTracer.ts

```diff
@@ -77,8 +77,8 @@ export class BasicTracer implements types.Tracer {
       ? TraceOptions.SAMPLED
       : TraceOptions.UNSAMPLED;
     const spanContext = { traceId, spanId, traceOptions, traceState };
-
-    if (!samplingDecision) {
+    const recordEvents = options.isRecordingEvents || false;
+    if (!recordEvents && !samplingDecision) {
```

Comment:
If `isRecordingEvents` is `true` and the sampling decision is `false`, this will cause the trace to be sampled. Is this the expected behavior? I'm also not sure I understand why this option is needed on `startSpan` specifically since my understanding is that it's an internal state that should not be directly configurable.

Reply:
> If isRecordingEvents is true and the sampling decision is false, this will cause the trace to be sampled. 

No, the trace will be still unsampled and these `span`s won't get exported to any backend. But you can use **z-pages** to view these span for in-process diagnostics. 

About z-pages:
OpenCensus provides in-process web pages that display collected data from the process that they are attached to. These are called “zPages”. They are useful to for in-process diagnostics without having to depend on any backend to examine traces or metrics. zPages can be useful during the development time or when the process to be inspected is known in production.